### DISCUSSION TITLE
What are the .tree files for?

### DISCUSSION THREAD

**Discussion by urataps** [PaulRBerg/prb-math#166] [Q&A] [answered]
I saw the .tree files in the tests and I wonder what are they for. Is it for informational purposes only? 
They look like hardhat style `context` and `it` callbacks for tests.

Are they also generated somehow? 
The output looks like one from the `tree` command in linux, but I know it is being used for viewing a directory structure, not for describing test requirements haha.

**Comment by PaulRBerg** [comment, marked as answer]
Hey! Great question.

The `*.tree` files are not for informational purposes. They are a key component of my strategy for writing clear, sound, and error-proof tests. You are right that the goal was to replicate the JavaScript `context` and `it` callbacks.

Recommended further reading:

- [Organize your unit tests by using a state tree](https://twitter.com/PaulRBerg/status/1624763320539525121)
- [Take your Solidity testing to the next level with state building!](https://sabnock.substack.com/p/take-your-solidity-testing-to-the)
